### PR TITLE
Improve indiduino driver reliability

### PR DESCRIPTION
**Pull request by nadvornik**
This pull request contains various fixes to ensure that indiduino driver shows real state of the device.

It is my first attempt to hack indi, so feedback is welcome.

### CODE REVIEW

**Comment by nadvornik** [comment]
changed as requested...

File: 3rdparty/indi-duino/libfirmata/src/firmata.cpp [hunk 1]
```diff
@@ -234,31 +251,67 @@ int Firmata::init(const char *_serialPort, uint32_t baud)
...
-        if (strlen(firmata_name) > 0)
-        {
-            if (debug)
-                printf("FIRMATA ARDUINO BOARD:%s\n", firmata_name);
-            fflush(stdout);
-            portOpen = 1;
-            break;
-        }
+        if (debug)
+            fprintf(stderr, "sf->openPort(fd) failed: exiting\n");
```
Comment (knro) [inline 3rdparty/indi-duino/libfirmata/src/firmata.cpp]: instead of if (debug)..etc, just use
LOGF_DEBUG

File: 3rdparty/indi-duino/indiduino.cpp [hunk 2]
```diff
@@ -549,20 +633,26 @@ bool indiduino::Connect()
...
+        }
+        LOG_INFO("Arduino board connected.");
+        LOGF_INFO("FIRMATA version:%s", sf->firmata_name);
+        IDSetSwitch(getSwitch("CONNECTION"), "CONNECTED. FIRMATA version:%s\n", sf->firmata_name);
         if (!setPinModesFromSKEL())
         {
-            IDLog("FAIL TO MAP ARDUINO PINS. CHECK SKELETON FILE SINTAX\n");
-            IDSetSwitch(getSwitch("CONNECTION"), "FAIL TO MAP ARDUINO PINS. CHECK SKELETON FILE SINTAX\n");
+            LOG_ERROR("Failed to map Arduino pins, check skeleton file syntax.");
+            IDSetSwitch(getSwitch("CONNECTION"), "Failed to map Arduino pins, check skeleton file syntax.\n");
```
Comment (knro) [inline 3rdparty/indi-duino/indiduino.cpp]: Remove \n

File: 3rdparty/indi-duino/libfirmata/src/firmata.cpp (hunk 1 above)
Comment (knro) [reply 3rdparty/indi-duino/libfirmata/src/firmata.cpp]: I see many instances of if (debug) and they should all be replaced accordingly.

**Comment by nadvornik** [comment]
Using INDI Logger in firmata is a bit tricky, because firmata library is used also standalone, without INDI.
This should however work.

**Pull request merged**